sdk/metric/aggregator/minmaxsumcount/mmsc.go

```diff
@@ -172,22 +142,14 @@ func (c *Aggregator) Merge(oa export.Aggregator, desc *metric.Descriptor) error
 		return aggregator.NewInconsistentMergeError(c, oa)
 	}
 
-	// Lock() synchronizes Merge() and Checkpoint() to ensure all operations of
-	// Merge() are performed on the same state.
-	c.lock.Lock()
-	defer c.lock.Unlock()
-
-	current := c.checkpoint()
-	ocheckpoint := o.checkpoint()
-
-	current.count.AddNumber(core.Uint64NumberKind, ocheckpoint.count)
-	current.sum.AddNumber(desc.NumberKind(), ocheckpoint.sum)
+	c.checkpoint.count.AddNumber(core.Uint64NumberKind, o.checkpoint.count)
```

Comment:
Shouldn't this still be guarded by a lock?

Reply:
I was confused on this.  Prior to the StateLocker approach, it was not protected by a mutex, and the docs for `Merge()` in the Aggregator interface say

```
// Merge() is called in a single-threaded context, no locking
// is required.
```
I believe the short summary is that once `Checkpoint()` is called on an aggregator, any `Merge()` calls to the same aggregator will happen before a subsequent `Checkpoint()` is ever called (and happen in the Collect phase of the batcher/integrator).

Someone please correct me if I'm wrong.

Reply:
Digging a bit into this, it seems like this assumption is built on there only ever being a single `Batcher`, which sounds like a good assumption to me. Looks good to me :+1:

Reply:
This is a true statement about the code today. We shouldn't need locking in Merge, because it's single-threaded.